packages/next/src/server/lib/router-utils/filesystem.ts

```diff
@@ -150,7 +150,14 @@ export async function setupFsCheck(opts: {
 
   if (!opts.dev) {
     const buildIdPath = path.join(opts.dir, opts.config.distDir, BUILD_ID_FILE)
-    buildId = await fs.readFile(buildIdPath, 'utf8')
+    try {
+      buildId = await fs.readFile(buildIdPath, 'utf8')
+    } catch (err: any) {
+      if (err.code !== 'ENOENT') throw err
+      throw new Error(
+        `Missing ${BUILD_ID_FILE}. Run \`next build\` before using \`next start\`, or use \`next dev\` for development.`
+      )
+    }
```

Comment:
Let's match the existing error from here https://github.com/vercel/next.js/blob/87015aceadf4d8480d768f739973cb723f090af4/packages/next/src/server/next-server.ts#L396-L407

Reply:
[60920f4](https://github.com/vercel/next.js/pull/62404/commits/60920f4be754cf47d4718f3b50544890a2a19359)